`CUDA_ARCH` not respected in GNU Make

I've encountered a compilation error when building AMReX-Castro on one HPC machine:


```
Loading ../../../external/amrex/Tools/GNUMake/comps/gnu.mak...
Loading ../../../external/amrex/Tools/GNUMake/comps/nvcc.mak...
Loading ../../../external/amrex/Tools/GNUMake/sites/Make.unknown...
../../../external/amrex/Tools/C_scripts/describe_sources.py \
          --git_names "Castro AMReX Microphysics" \
          --git_dirs "../../.. ../../../external/amrex ../../../external/Microphysics"

source git hashes:
  Castro              : 25.04-24-g99e9bc037-dirty
  AMReX               : 25.04-dirty
  Microphysics        : 25.04-dirty


Compiling Castro.cpp ...
mpicxx -MMD -MP -ccbin=g++ -Xcompiler='-Werror=return-type -gdwarf-4 -O3 -finline-limit=43210 -std=c++17  -pthread  -std=c++17' --std=c++17 -Wno-deprecated-gpu-targets -m64 -maxrregcount=255    --expt-relaxed-constexpr --expt-extended-lambda --forward-unknown-to-host-compiler -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -Xcudafe --diag_suppress=implicit_return_from_non_void_function --Werror cross-execution-space-call -lineinfo --ptxas-options=-O3 --use_fast_math  --Werror ext-lambda-captures-this --display-error-number --diag-error 20092 --generate-code arch=compute_5350,code=sm_5350 -x cu -dc   -DBL_LAZY -DAMREX_LAZY -DBL_USE_MPI -DAMREX_USE_MPI -DAMREX_MPI_THREAD_MULTIPLE -DBL_NO_FORT -DAMREX_USE_CUDA -DAMREX_USE_GPU -DBL_COALESCE_FABS -DAMREX_GPU_MAX_THREADS=256    -DAMREX_USE_GPU_RDC -DBL_SPACEDIM=2 -DAMREX_SPACEDIM=2 -DBL_FORT_USE_UNDERSCORE -DAMREX_FORT_USE_UNDERSCORE -DBL_Linux -DAMREX_Linux -DAMREX_DIMENSION_AGNOSTIC -DNDEBUG -DAMREX_NO_PROBINIT -DOMPI_SKIP_MPICXX -DSPONGE -DSTRANG -DSCREEN_METHOD=SCREEN_METHOD_screen5 "-DNAUX_NET=0" -DAUTODIFF_DEVICE_FUNC=AMREX_GPU_HOST_DEVICE -Itmp_build_dir/s/2d.gnu.MPI.CUDA.EXE -I. -I../../../external/amrex/Src/Base -I../../../external/amrex/Src/Base/Parser -I../../../external/amrex/Src/AmrCore -I../../../external/amrex/Src/Amr -I../../../external/amrex/Src/Boundary -I../../../external/Microphysics/util -I../../../external/Microphysics/util/autodiff -I../../../external/Microphysics/util/gcem/include -I./ -I../../../Source/driver -I../../../Source/hydro -I../../../Source/problems -I../../../Source/sources -I../../../external/Microphysics/EOS -I../../../external/Microphysics/EOS/gamma_law -I../../../external/Microphysics/networks/general_null -I../../../external/Microphysics/EOS -I../../../external/Microphysics/networks -I../../../external/Microphysics/interfaces -I../../../external/Microphysics/constants -I../../../external/Microphysics/util/approx_math -Itmp_build_dir/castro_sources/2d.gnu.MPI.CUDA.EXE -I../../../external/amrex/Tools/C_scripts -isystem /usr/local/cuda/include -I/usr/lib/x86_64-linux-gnu/openmpi/include -I/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi -c ../../../Source/driver/Castro.cpp -o tmp_build_dir/o/2d.gnu.MPI.CUDA.EXE/Castro.o
nvcc fatal   : Unsupported gpu architecture 'compute_5350'
make: *** [../../../external/amrex/Tools/GNUMake/Make.rules:262: tmp_build_dir/o/2d.gnu.MPI.CUDA.EXE/Castro.o] Error 1
```

Upon investigation, I found that the issue was caused by incorrect GPU architecture detection: the machine I'm using has `deviceQuery` in an unconventional path (`/lsc/opt/cuda-12.5/extras/demo_suite/deviceQuery`), and `nvidia-smi` fails for some reason:

```
$ nvidia-smi
Failed to initialize NVML: Driver/library version mismatch
NVML library version: 535.247
```

The `535` in the `nvidia-smi` output was parsed as the GPU architecture, causing the compilation problem.

I therefore tried to override the architecture by setting `CUDA_ARCH = 80`, but [it was overridden in `Tools/GNUMake/sites/Make.unknown`](https://github.com/AMReX-Codes/amrex/blob/788601034a9ad730d44b1e40da9d3f6f66f21f08/Tools/GNUMake/sites/Make.unknown#L104):

``` Make
... (omitted)

ifneq ($(NO_CUDA_CHECKING),TRUE)
ifeq ($(USE_CUDA),TRUE)

  ... (omitted)

  ifneq ($(cuda_device_query_result),)

     ... (omitted)

     ifeq ($(TEMP_CUDA_ARCH),3.5)
         CUDA_ARCH = 35
     else
         CUDA_ARCH = $(firstword $(subst ., ,$(TEMP_CUDA_ARCH)))0
     endif
  endif
endif
endif
```

It would be better if `CUDA_ARCH` were not overridden when the user specifies it.

--- a/Tools/GNUMake/sites/Make.unknown
+++ b/Tools/GNUMake/sites/Make.unknown
@@ -98,13 +98,7 @@ ifeq ($(USE_CUDA),TRUE)
   endif
   ifneq ($(cuda_device_query_result),)
      TEMP_CUDA_ARCH = $(lastword $(cuda_device_query_result))
-#      CUDA_ARCH = $(subst .,,$(TEMP_CUDA_ARCH))
-     # We do this because pgfortran does not support all cuda arches.
-     ifeq ($(TEMP_CUDA_ARCH),3.5)
-         CUDA_ARCH = 35
-     else
-         CUDA_ARCH = $(firstword $(subst ., ,$(TEMP_CUDA_ARCH)))0
-     endif
+     CUDA_ARCH = $(subst .,,$(TEMP_CUDA_ARCH))
   endif
 endif
 endif

--- a/Tools/GNUMake/sites/Make.unknown
+++ b/Tools/GNUMake/sites/Make.unknown
@@ -94,7 +94,7 @@ ifeq ($(USE_CUDA),TRUE)
      cuda_device_query_result := $(shell $(CUDA_HOME)/extras/demo_suite/deviceQuery | grep "CUDA Capability")
   else
      # try nvidia-smi, as deviceQuery is no longer distributed with CUDA 12 and up
-     cuda_device_query_result := $(shell nvidia-smi --query-gpu=compute_cap --format=csv,noheader)
+     cuda_device_query_result := $(shell nvidia-smi --query-gpu=compute_cap --format=csv,noheader || echo $(CUDA_ARCH))
   endif
   ifneq ($(cuda_device_query_result),)
      TEMP_CUDA_ARCH = $(lastword $(cuda_device_query_result))

